src/katex.less

```diff
@@ -635,3 +635,23 @@
     text-align: left;
     padding-left: 2em;
 }
+
+// Automatic equation numbers for some environments.
+// Use parallel counters for HTML and MathML.
+body {
+    counter-reset: katexEqnNo mmlEqnNo;
+}
+
+.eqn-num::before {
+    counter-increment: katexEqnNo;
+    content: "(" counter(katexEqnNo) ")";
+}
+
+.mml-eqn-num::before {
+    counter-increment: mmlEqnNo;
+    content: "(" counter(mmlEqnNo) ")";
+}
+
+.mtr-glue {
+    width: 50%;
+}
```

Comment:
I think these should be a `contrib` extension for now.

Reply:
I agree that we should be careful with CSS counters. First, just to make sure that you and I understand each other, let me describe what I think is the potential problem. It lies in the CSS code `body { counter-reset: katexEqnNo mmlEqnNo; }`. The trouble is that a counter-reset statement can over-write all counters. This line of CSS would either over-write a page's existing counters or be over-written, depending on the order in which CSS files are loaded.

Now let me suggest another approach.

1. Remove from this PR the CSS line `body { counter-reset: katexEqnNo mmlEqnNo; }`.
2. Let the other CSS in this PR stand. It does no harm.
3. Amend the setup documentation to strongly suggest that web site administrators should include the `body { counter-reset: katexEqnNo mmlEqnNo; }` CSS in their page.
4. For convenience, create a `contrib` CSS file that has only one line, that being `body { counter-reset: katexEqnNo mmlEqnNo; }`.
5. Test this PR without a counter-reset and make sure that the worst outcome is the omission of automatic numbering and perhaps some console messages.

Am I addressing the same problem that you see? What do you think of my suggestion?

Reply:
I was thinking moving all styles related to numbering out to a `contrib` module, but now I think it'd make KaTeX hard to configure. We could rather add an option to choose which numbering to use, e.g., CSS counter, Javascript, none, etc. (I'll send a PR later) I think `body { counter-reset: katexEqnNo mmlEqnNo; }` wouldn't cause too much trouble, so I think it's okay to leave it here.

Other styles, `.eqn-num::before`, `.mml-eqn-num::before`, and `.mtr-glue`, should be scoped, i.e., inside `.katex { ... }`.

Reply:
Agree 100% on scoping the styles inside `.katex{...}`. On it.

Reply:
Styles are scoped inside `.katex{...}`.

If we let the CSS counter-reset remain as is, then I think the KaTeX setup documentation should be amended. That can perhaps be a later PR. I would like all the automatic numbering environments to go into the same KaTeX release.